Changes for adding mcp tool method as a decorator

Jayssgss opened the pull request.

Jayssgss committed:
Changes for adding mcp tool method as a decorator

Jayssgss committed:
Update with latest code

gyliu513 commented on docs/PRIVACY.md:
```diff
@@ -0,0 +1,31 @@
+# Privacy Policy
```
Your PR should not contain this, it was merged at https://github.com/instana/mcp-instana/pull/34

gyliu513 commented on mcpb/manifest.json:
```diff
@@ -0,0 +1,100 @@
+{
+  "manifest_version": "0.2",
+  "name": "mcp-instana",
+  "display_name": "MCP Instana Server",
```
This was merged at https://github.com/instana/mcp-instana/pull/35

gyliu513 commented on src/core/server.py:
```diff
@@ -151,7 +151,18 @@ def create_app(token: str, base_url: str, port: int = int(os.getenv("PORT", "808
-                        server.tool()(bound_method)
+
+                        # Check if the method has MCP metadata
+                        if hasattr(bound_method, '_mcp_title') and hasattr(bound_method, '_mcp_annotations'):
```
I think the `if` checking will be finally removed after you add  annotation to all tools, right?

gyliu513 commented on src/core/utils.py:
```diff
@@ -10,13 +10,45 @@
+# Import MCP dependencies
+from mcp.types import ToolAnnotations
+
+from src.prompts import mcp
```
why this import?

Jayssgss committed:
Merge remote-tracking branch 'upstream/main' into mcp-tools-as-decorator

Jayssgss committed:
Update @register_as_tool with with server.tools as server.tools's wrapper

gyliu513 reviewed (commented):
Please do not touch files under `docs` folder.

Make this PR just focus on the decorator, do not touch other code.

Jayssgss committed:
restore: add back docs/PRIVACY.md and docs/assets/ from previous commit

- Restore PRIVACY.md file that was accidentally removed
- Restore docs/assets/ folder with .gitattributes and instana.svg
- Keep docs folder updated as requested by Guangya
